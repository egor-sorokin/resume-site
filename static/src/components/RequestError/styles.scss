@import '../../styles/base/_vars.scss';

.request-error {
  width: 100vw;
  height: 100vh;

  background-color: $mercury;

  &__inner {
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;

    height: 100%;
  }
  &__text {
  }
  &__link {
    transition: color $defaultTiming $defaultEasing;

    &:hover {
      color: $duneLight;
    }
  }
}
