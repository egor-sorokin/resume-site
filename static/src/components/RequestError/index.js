import React from 'react';
import './styles.scss';

const RequestError = () => (
  <div class="request-error">
    <div class="request-error__inner">
      <h3 class="request-error__text text-c-l-dune">
        Something went wrong, please contact
        <a
          href="https://www.linkedin.com/in/egor-sorokin-1b13ba105"
          target="_blank"
          class="request-error__link"
        > administrator</a>
      </h3>
    </div>
  </div>
);

export default RequestError;
