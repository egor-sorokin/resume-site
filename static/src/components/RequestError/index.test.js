// eslint-disable-next-line 
import {shallow, render, mount} from 'enzyme';
import React from 'react';
import RequestError from './index';


describe('<RequestError />', () => {
  let component;

  describe('base tests', () => {
    beforeEach(() => {
      component = shallow(<RequestError />);
    });


    it('renders without crashing', () => {
      expect(component).toBeDefined();
    });


    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });


    it('renders 1 component', () => {
      expect(component.children()).toHaveLength(1);
    });
  });
}); 
