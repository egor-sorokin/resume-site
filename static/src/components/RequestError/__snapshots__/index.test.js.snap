// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestError /> base tests matches snapshot 1`] = `
<div
  class="request-error"
>
  <div
    class="request-error__inner"
  >
    <h3
      class="request-error__text text-c-l-dune"
    >
      Something went wrong, please contact
      <a
        class="request-error__link"
        href="https://www.linkedin.com/in/egor-sorokin-1b13ba105"
        target="_blank"
      >
         administrator
      </a>
    </h3>
  </div>
</div>
`;
